src/components/PopupWithForm.js

```diff
@@ -22,6 +22,18 @@ export class PopupWithForm extends Popup {
     return inputValues;
   }
 
+  setLoading(isLoading) {
+    if (isLoading) {
+      this._submitButton.textContent = "Saving...";
+    } else {
+      this._submitButton.textContent = this._originalSubmitText;
+    }
+  }
+
+  setSubmit(handleSubmit) {
+    this._handleFormSubmit = handleSubmit;
+  }
+
```

Comment:
Please, delete `this._popupForm.reset();` from line 46

You need to **reset** the form   in block `then` in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.